jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/impl/util/HumanTaskHandlerHelper.java

```diff
@@ -201,6 +216,72 @@ public static Deadlines setDeadlines(Map<String, Object> parameters, List<Organi
 		return notifications;
 	}
 
+	protected static int getDeadlineRepeatLimit(String deadlineStr) {
+		String[] repeatableParts = DateTimeUtils.parseISORepeatable(deadlineStr);
+		String repeatLimit = repeatableParts[0];
+		if (!repeatLimit.isEmpty()) {
+			return Integer.parseInt(repeatLimit);
+		} else {
+			return -1;
+		}
+	}
+
+	protected static Date getDeadlineDurationDate(String durationStr, int repeatCount) {
+		// handles iso repetable and both period and date notation
+		try {
+
+			if (DateTimeUtils.isRepeatable(durationStr)) {
+				String[] repeatableParts = DateTimeUtils.parseISORepeatable(durationStr);
+				String tempTimeDelay = repeatableParts[1];
+				String tempDateTimeStr = repeatableParts[2];
+
+				Date retDate;
+
+				if (DateTimeUtils.isPeriod((tempDateTimeStr))) {
+					// e.g R/start/duration
+					retDate = new Date(DateTimeUtils.parseDateTime(tempTimeDelay) + repeatDuration(tempDateTimeStr, repeatCount));
+				} else {
+					if(DateTimeUtils.isPeriod(tempTimeDelay)) {
+						// e.g R/duration/end
+						long firstRet = DateTimeUtils.parseDateTime(tempDateTimeStr) -  Duration.parse(tempTimeDelay).toMillis();
+						retDate =  new Date(firstRet + repeatDuration(tempTimeDelay, repeatCount));
+					} else {
+						// e.g R/start/end
+						// duration is end - start
+						long duration = DateTimeUtils.parseDateTime(tempDateTimeStr) - DateTimeUtils.parseDateTime(tempTimeDelay);
+						retDate =  new Date(DateTimeUtils.parseDateTime(tempTimeDelay) + repeatDuration(duration, repeatCount));
+					}
+				}
+
+				return retDate;
+			} else {
+				if (DateTimeUtils.isPeriod((durationStr))) {
+					return new Date(System.currentTimeMillis() + Duration.parse(durationStr).toMillis());
+				} else {
+					return new Date(System.currentTimeMillis() + DateTimeUtils.parseDateAsDuration(durationStr));
+				}
+			}
+		} catch(Exception e) {
+			throw new IllegalArgumentException("Unable to parse duration string: " + durationStr + " : " + e.getMessage());
+		}
+
+	}
+
+	protected static long repeatDuration(String durationStr, int repeatCount) {
+		long duration = Duration.parse(durationStr).toMillis();
+		for(int i = 0; i < repeatCount; i++) {
+			duration += duration;
+		}
+		return duration;
+	}
```

Comment:
Here when `repeatCount` is 0, we will add duration in `new Date(DateTimeUtils.parseDateTime(tempTimeDelay) + repeatDuration(tempDateTimeStr, repeatCount));` anyway. But as I wrote in the email, first trigger should occur at the beginning of the interval. Subsequent triggers should be then shifted by a duration.

So I think we need here just `return Duration.parse(durationStr).toMillis()*repeatCount`. If it is 0, it won't add anything. E.g. with R3 we will have one trigger at the startDate, then startDate+duration, then startDate+2*duration.